src/main/java/com/sap/sgs/phosphor/fosstars/model/score/AverageCompositeScore.java

```diff
@@ -0,0 +1,120 @@
+package com.sap.sgs.phosphor.fosstars.model.score;
+
+import static com.sap.sgs.phosphor.fosstars.model.other.Utils.setOf;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.sap.sgs.phosphor.fosstars.model.Feature;
+import com.sap.sgs.phosphor.fosstars.model.Score;
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.value.ScoreValue;
+import com.sap.sgs.phosphor.fosstars.model.value.ValueHashSet;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A base class for scores which are based on an average of other scores (sub-scores).
+ */
+public class AverageCompositeScore extends AbstractScore {
+
+  /**
+   * A set of sub-scores.
+   */
+  private final Set<Score> subScores;
+
+  /**
+   * Initializes a new {@link AverageCompositeScore} with a number of sub-scores.
+   *
+   * @param name A name of the score.
+   * @param subScores A number of sub-scores.
+   */
+  public AverageCompositeScore(String name, Score... subScores) {
+    this(name, setOf(subScores));
+  }
+
+  /**
+   * Initializes a new {@link AverageCompositeScore} with a set of sub-scores.
+   *
+   * @param name A name of the score.
+   * @param subScores A set of sub-scores.
+   */
+  @JsonCreator
+  AverageCompositeScore(
+      @JsonProperty("name") String name,
+      @JsonProperty("subScores") Set<Score> subScores) {
+
```

Comment:
Please remove this line.

Reply:
I would prefer to add an extra line when parameters of a method take multiple lines. The extra line helps to separate the method definition from the body.

Reply:
ok